011-kafka-canary.md

```diff
@@ -0,0 +1,123 @@
+# Strimzi Canary
+
+Implement a Kafka canary tool which will act as an indicator of whether Kafka clusters are operating correctly.
+This would be achieved by creating a canary topic and periodically producing and consuming events on the topic. 
+
+## Current situation
+
+Currently Strimzi does not have any canary style feature.
+It does have health checks for the Kafka cluster which can be replaced/augmented with the canary.
+
+## Motivation
+
+The canary would provide the following in indication that the Kafka cluster is operating as expected from a users perspective. I.e. messages can successfully be produced and consumed. 
+Some would be exported to reflect the canary's activity. Initial metrics would included:
+  * records-produced-total
+  * records-consumed-total
+  * produce-error
+  * consume-error
+  * latency
+
+In future the canary can be used by the Kafka Roller to improve health checks. 
+However this is beyond the scope of the current proposed work.
+
+## Proposal
+
+The canary would be run as a separate pod alongside Strimzi operator. 
```

Comment:
AFAIU the canary will run per Kafka cluster ... so maybe _alongside the Kafka cluster_ instead of _alongside Strimzi operator_?